build error on master under Windows

```
D:\legalize\llvm\llvm-project\llvm\include\llvm\Support\DXILABI.h(27,3): error C2144: syntax error: 'void' should be preceded by '}' [D:\legalize\llvm\build-iwyu\tools\iwyu\include-what-you-use.vcxproj]
  (compiling source file '../../../include-what-you-use/iwyu_ast_util.cc')
```

Line 27 of `DXILABI.h` contains the following:
```
enum class ParameterKind : uint8_t {
  INVALID = 0,
  VOID,  // line 27
  HALF,
  FLOAT,
  DOUBLE,
  I1,
  I8,
  I16,
  I32,
  I64,
  OVERLOAD,
  CBUFFER_RET,
  RESOURCE_RET,
  DXIL_HANDLE,
};
```

A Windows header files define `VOID` as a macro that expands to `void`.  The offending header is the inclusion of `Shlwapi.h` on line 78 of `iwyu_port.h`.

Hacking in a simple `#undef VOID` after the inlusion of that header results in a clean build.

Branch `clang_18` of include-what-you-use with branch `release/18.x` of llvm/clang builds without errors.

[iwyu-build-log.txt](https://github.com/include-what-you-use/include-what-you-use/files/14608595/iwyu-build-log.txt)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,6 +110,7 @@ add_llvm_executable(include-what-you-use
   iwyu_location_util.cc
   iwyu_output.cc
   iwyu_path_util.cc
+  iwyu_port.cc
   iwyu_preprocessor.cc
   iwyu_regex.cc
   iwyu_verrs.cc

--- a/iwyu_port.cc
+++ b/iwyu_port.cc
@@ -1,0 +1,26 @@
+//===--- iwyu_port.cc - portability shims ---------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#if defined(_WIN32)
+
+#include "Shlwapi.h"  // for PathMatchSpecA
+
+bool GlobMatchesPath(const char *glob, const char *path) {
+  return PathMatchSpecA(path, glob);
+}
+
+#else  // #if defined(_WIN32)
+
+#include <fnmatch.h>
+
+bool GlobMatchesPath(const char *glob, const char *path) {
+  return fnmatch(glob, path, 0) == 0;
+}
+
+#endif  // #if defined(_WIN32)

--- a/iwyu_port.h
+++ b/iwyu_port.h
@@ -64,28 +64,6 @@ class OstreamVoidifier {
   ::include_what_you_use::FatalMessageEmitter( \
       __FILE__, __LINE__, message).stream()
 
-#if defined(_WIN32)
-
-#define NOMINMAX 1 // Prevent Windows headers from redefining min/max.
-#include "Shlwapi.h"  // for PathMatchSpecA
-
-// This undef is necessary to prevent conflicts between llvm
-// and Windows headers.
-// objbase.h has #define interface struct.
-#undef interface
-
-inline bool GlobMatchesPath(const char *glob, const char *path) {
-  return PathMatchSpecA(path, glob);
-}
-
-#else  // #if defined(_WIN32)
-
-#include <fnmatch.h>
-
-inline bool GlobMatchesPath(const char *glob, const char *path) {
-  return fnmatch(glob, path, 0) == 0;
-}
-
-#endif  // #if defined(_WIN32)
+bool GlobMatchesPath(const char *glob, const char *path);
 
 #endif  // INCLUDE_WHAT_YOU_USE_PORT_H_

